### PR TITLE
fix selectLocation action

### PR DESCRIPTION
**Pull request by codehag**
Associated Issue: #4844 

This fixes the inconsistency and the test. however it also seems to have fixed pretty print, which was not working and not noticed until now. the CI tests are failing, they need to be updated to reflect that we pretty print automatically now. will revisit in the morning!

### CODE REVIEW

File: src/actions/sources.js [hunk 1]
```diff
@@ -322,7 +321,7 @@ export function closeTabs(urls: string[]) {
     const sourceId = getNewSelectedSourceId(getState(), tabs);
 
     dispatch({ type: "CLOSE_TABS", urls, tabs });
-    dispatch(selectLocation({ id: sourceId }));
+    dispatch(selectLocation({ sourceId, line: 0 }));
```
Comment (codehag) [inline src/actions/sources.js]: this is a left over

File: src/components/Editor/Tabs.js [hunk 2]
```diff
@@ -320,7 +320,8 @@ class SourceTabs extends PureComponent<Props, State> {
     const { selectLocation } = this.props;
     const filename = getFilename(source.toJS());
 
-    const onClick = () => selectLocation({ id: source.get("id") });
+    const onClick = () =>
+      selectLocation({ sourceId: source.get("id"), line: 0 });
```
Comment (codehag) [inline src/components/Editor/Tabs.js]: leftover

**Comment by jasonLaster** [comment]
Thanks @codehag!

Lets see if we can tighten up our flow types to catch this. Maybe Location should be a strict type `{| |}`.

File: src/components/QuickOpenModal.js [hunk 3]
```diff
@@ -140,17 +140,16 @@ export class QuickOpenModal extends Component<Props, State> {
       if (selectedSource == null) {
         return;
       }
-      selectLocation({
-        id: selectedSource.get("id"),
-        ...(item.location != null ? { line: item.location.start.line } : null)
-      });
+      const line =
```
Comment (lukaszsobek) [inline src/components/QuickOpenModal.js]: The 0 part would be covered at component mount time by https://github.com/devtools-html/debugger.html/pull/4836 , I'd rather null this here so as not to have this functionality hardcoded in too many places.

File: src/components/QuickOpenModal.js [hunk 4]
```diff
@@ -140,17 +140,16 @@ export class QuickOpenModal extends Component<Props, State> {
...
     } else if (searchType === "gotoSource") {
       const location = parseLineColumn(query);
       if (location != null) {
-        selectLocation({ id: item.id, ...location });
+        selectLocation({ ...location, sourceId: item.id });
       }
     } else {
-      selectLocation({ id: item.id });
+      selectLocation({ sourceId: item.id, line: 0 });
     }
```
Comment (lukaszsobek) [inline src/components/QuickOpenModal.js]: Same as above.

File: src/components/QuickOpenModal.js [hunk 5]
```diff
@@ -168,10 +167,9 @@ export class QuickOpenModal extends Component<Props, State> {
...
 
     if (searchType === "variables") {
-      selectLocation({
-        id: selectedSource.get("id"),
-        ...(item.location != null ? { line: item.location.start.line } : null)
-      });
+      const line =
+        item.location && item.location.start ? item.location.start.line : 0;
```
Comment (lukaszsobek) [inline src/components/QuickOpenModal.js]: here too

**Comment by lukaszsobek** [comment]
@codehag @jasonLaster 
- Tests update pending via pr on codehag's repo
- We are using` Location` as type in several places, it requires i.a. line explicitly. selectLocation (which is currently using `Location`) on the other hand is perfectly fine without having a location set. The problem is that is passes the location on directly, so dependent functions will not get their line if it wasn't already set. Some solutions might be:
1.) give `selectLocation` its own type that does not require line nor column, but inside the function make sure line and column are set to 0 if not provided.
2.) make every `selectLocation` call have line explicitly set to 0 if there is nothing else

**Pull request merged**